script.js

```diff
@@ -1,4 +1,200 @@
+let playerDices = [];
+let autoArrange = [];
+let numOfDices = 0;
+let playersArray;
+let nDiceMode = false;
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  // on submit dice is rolled
+  let message;
+  let resultedNumber;
+  if (mode == 'set num of players') {
+    numOfPlayers = input;
+    mode = 'num of dices';
+    return `Number of players set to ${numOfPlayers} <p> Player1 is the default first player. <p> Play basic (2 dices), press submit <p> or Enter number of dices to play n - dice mode. `;
+  }
+  //
+  if (mode == 'num of dices') {
+    numOfDices = input;
+    // mode = 'playing';
+    if (numOfDices != '') {
+      mode = 'n dices';
+      nDiceMode = true;
+      return `Number of Dices set to ${numOfDices} <p> Player1 is the default first player. <p> press 'submit' to start playing`;
+    }
+    if (numOfDices == 0) {
+      mode = 'playing';
+      return 'play in normal mode. hit submit to start.';
+    }
+  }
+
+  if (roundsPlayed < numOfPlayers) {
+    if (mode == 'playing') {
+      firstRoll = rollDice();
+      secondRoll = rollDice();
+      message = `Welcome ${currentTurn}. <p> You rolled ${firstRoll} for Dice 1 and ${secondRoll} for Dice 2.<p> type 'manual' or 'auto' to chose mode to generate number.`;
+      mode = 'choosing';
+      return message;
+    }
+    if (mode == 'n dices') {
+      playersArray = rollDiceNTimes(numOfDices);
+      message = `Your dice rolls are ${playersArray}. <p> input the indexes to order the numbers`;
+      mode = 'input indexes';
+      return message;
+    }
+    if (mode == 'input indexes') {
+      let inputIndexes = input;
+      resultedNumber = generateNumFromIndex(inputIndexes, playersArray);
+      console.log('resulted number ' + resultedNumber);
+    }
+    if (mode == 'choosing') {
+      mode = setMode(input);
+      if (mode == 'manual') {
+        return `${mode} is set. <p> Choose the order of the dice (i.e dice1 or dice2).`;
+      }
+      return `${mode} is set. <p> press 'submit' to see auto chosen number.`;
+    }
+    if (mode == 'manual') {
+      console.log('choosing mode works');
+      resultedNumber = arrangeDice(input, firstRoll, secondRoll);
+    }
+    if (mode == 'auto') {
+      autoArrange.push(firstRoll);
+      autoArrange.push(secondRoll);
+      resultedNumber = autoGenNumber(autoArrange);
+      autoArrange = [];
+    }
+    playerDices.push(resultedNumber);
+    message = `your score is ${playerDices[roundsPlayed]}.`;
+    roundsPlayed += 1;
+    if (roundsPlayed == numOfPlayers) {
+      message += 'The game round is over.<p> Enter \'lowest\' to see result in lowest score mode <p> Or press submit to see the winner in normal mode(highest score mode)';
+      return message;
+    }
+    currentTurn = 'player' + (roundsPlayed + 1);
+    message += `next player is ${currentTurn}`;
+    mode = 'playing';
+    if (nDiceMode) {
+      mode = 'n dices';
+    }
+    return message;
+  }
```

Comment:
Can some of this logic be abstracted into a helper function to simplify the `main` function? Otherwise `main` function seems very long.